R/util-MakeSnapshotHelpers.R

```diff
@@ -330,3 +330,129 @@ MakeRptKRIBoundsDetails <- function(lResults, config_workflow, gsm_analysis_date
 }
 
 
+#' Create rpt_qtl_threshold_param output for `Make_Snapshot()`
+#'
+#' @param meta_param `data.frame` the meta_param defined in lMeta argument of `Make_Snapshot()` Default: gsm::meta_param
+#' @param status_param `data.frame` the config_param defined in lMeta argument of `Make_Snapshot()`
+#' @param gsm_analysis_date `string` Date of snapshot
+#' @param type `string` type of threshold to output
+#' @param verbose `logical` whether or not to display function messages
+#'
+#' @export
+#'
+#' @keywords internal
+MakeRptThresholdParam <- function(meta_param, status_param, gsm_analysis_date, type, verbose = FALSE){
+  if(!type %in% c("kri", "qtl")){
+    stop("`type` must be either 'kri' or 'qtl'")
+  }
+  ID <- paste0(type, "_id")
+  if( is.null(meta_param) & is.null(status_param) ) {
+    if(verbose) {cli::cli_alert_warning("No `meta_param` or `status_param` found, returning blank data frame.")}
+    data.frame("study_id" = NA,
+               "snapshot_date" = NA,
+               "workflowid" = NA,
+               "gsm_version" = NA,
+               "param" = NA,
+               "index_n" = NA,
+               "default_s" = NA,
+               "configurable" = NA,
+               "pt_cycle_id" = NA,
+               "pt_data_dt" = NA) %>%
+      rename_at("workflowid", ~paste0(type, "_id"))
+  }
+  if( is.null(meta_param) & !is.null(status_param) ) {
+    if(verbose) {cli::cli_alert_warning("`MakeRptQTLThresholdParam()` is missing meta_param, status_param will be used to define defaults")}
+    status_param %>%
+      filter(grepl(type, .data$workflowid)) %>%
+      mutate("study_id" = Study_Map_Raw()[["studyid"]],
```

Comment:
```suggestion
      mutate("study_id" = NA_character_,
```
I think we should return `NA` here rather than using default `clindata` values.